### PR TITLE
Unify naming convention in REST API

### LINKED ISSUE #1197: Make all REST API query params / json payload in snake_case
In Quickwit 0.2, the Search REST API is **camelCase**.

@PSeitz recently added aggregation with a compatible Elastic API that is **snake_case** AFAIK.

We should use *snake_case* for all query params / json payload to avoid inconsistencies.

### PR DESCRIPTION
**Pull request by dzvon**
Closes #1197
 
This PR will make query params, json payload and json response fields in *snake_case* naming convention.

Note: This is a BREAKING CHANGE.

### CODE REVIEW

**Comment by dzvon** [comment]
I'll update the related document later.

**Comment by fulmicoton** [comment]
Thank you @dzvon ! @fmassot can you review this?

File: quickwit-search/src/search_response_rest.rs [hunk 1]
```diff
@@ -26,7 +26,7 @@ use crate::error::SearchError;
 /// SearchResponseRest represents the response returned by the REST search API
 /// and is meant to be serialized into JSON.
 #[derive(Serialize)]
-#[serde(rename_all = "camelCase")]
+#[serde(rename_all = "snake_case")]
```
Comment (fmassot) [inline quickwit-search/src/search_response_rest.rs:29]: I think `#[serde(rename_all = "snake_case")]` is unnecessary as all rust variables are `snake_case`

Comment (fmassot) [reply quickwit-search/src/search_response_rest.rs:29]: Can you remove the `rename_all = "snake_case"` whenever it's unnecessary?

**Review by fmassot — CHANGES_REQUESTED** [review]
Almost good!  see comments.

File: quickwit-search/src/search_response_rest.rs (hunk 1 above)
Comment (dzvon) [reply quickwit-search/src/search_response_rest.rs:29]: Sure.

**Review by fulmicoton — APPROVED** [review]

**Review by fulmicoton — APPROVED** [review]

**Pull request merged**